src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -335,7 +355,14 @@ protected void OnDocumentContextUpdated(DocumentId documentId)
 
             if (container != null)
             {
-                OnDocumentContextUpdated(documentId, container);
+                if (_isProjectUnloading)
```

Comment:
This code is considered free threaded, which is why there is no threading asserts. It should be made safe.